lib/states/statesInRedis.js

```diff
@@ -105,12 +105,27 @@ class StateRedis {
         let connected = false;
         let reconnectCounter = 0;
 
-        if (this.settings.connection.port === 0) {
-            // initiate a unix socket connection using the parameter 'host'
-            this.client = new Redis(this.settings.connection.host, this.settings.connection.options);
+        if (this.settings.connection.port === 0) { // Port = 0 means unix socket
+            // initiate a unix socket connection
+            this.settings.connection.options.path = this.settings.connection.host;
+            this.log.debug('Redis States: Use File Socket for connection: ' + this.settings.connection.options.path);
+        } else if (Array.isArray(this.settings.connection.host)) { // Host is an array means we use a sentinel
+            const defaultPort = Array.isArray(this.settings.connection.port) ? null : this.settings.connection.port;
+            this.settings.connection.options.sentinels = [];
+            this.settings.connection.host.forEach((redisNode, idx) => {
+                this.settings.connection.options.sentinels.push({
+                    host: redisNode,
+                    port: defaultPort ? defaultPort : this.settings.connection.port[idx]
+                });
+            });
```

Comment:
`this.settings.connections.options.sentinels = this.settings.connections.host.map((redisNode, idx) => ({ ... }));`

Reply:
changed, hopefully correct